src/rgw/rgw_op.cc

```diff
@@ -3273,13 +3275,20 @@ static CompressorRef get_compressor_plugin(const req_state *s,
   const auto& upload_id = s->info.args.get("uploadId", &is_multipart);
 
   if (!is_multipart) {
-    return Compressor::create(s->cct, compression_type);
-  }
+    int count = 0;
+recreate:
+    compressor = Compressor::create(s->cct, compression_type);
+    if ((compressor == nullptr) && (++count <= 3))
+      goto recreate;
+    return compressor;
+ }
 
   // use a hash of the multipart upload id so all parts use the same plugin
-  const auto alg = std::hash<std::string>{}(upload_id) % Compressor::COMP_ALG_LAST;
+  int alg = std::hash<std::string>{}(upload_id) % Compressor::COMP_ALG_LAST;
   if (alg == Compressor::COMP_ALG_NONE) {
-    return nullptr;
+    alg = (std::hash<std::string>{}(upload_id) / Compressor::COMP_ALG_LAST) % Compressor::COMP_ALG_LAST;
+    if (alg == Compressor::COMP_ALG_NONE)
+      alg = Compressor::COMP_ALG_LAST - 1;
```

Comment:
@fangyuxiangGL Is there a reason to exclude 'none' compressor from list of selected compressors when 'random' is slelected ?